src/intro_movie_scene_4.c

```diff
@@ -0,0 +1,626 @@
+}
+
+static void IntroMovie_Scene4_DestroySprites(IntroMovieOvyData *data, IntroMovieScene4Data *sceneData) {
+    Sprite_Delete(sceneData->hand1Sprite);
+    Sprite_Delete(sceneData->hand2Sprite);
+    Sprite_Delete(sceneData->sparklesSprite);
+    for (u8 i = 0; i < 3; ++i) {
+        Sprite_Delete(sceneData->starterSprites[i]);
+    }
+    for (u8 i = 0; i < 4; ++i) {
+        sub_0200AEB0(sceneData->spriteResObjs[i][GF_GFX_RES_TYPE_CHAR]);
+        sub_0200B0A8(sceneData->spriteResObjs[i][GF_GFX_RES_TYPE_PLTT]);
+    }
+    IntroMovie_DestroySpriteResourceManagers(data);
+}
+
+static void IntroMovie_Scene4_CreateSprites(IntroMovieOvyData *data, IntroMovieScene4Data *sceneData) {
+    SpriteResourcesHeader header;
+    SpriteTemplate template;
+    int monSpriteResIds[3] = {1, 2, 3};
+
+    IntroMovie_BuildSpriteResourcesHeaderAndTemplate(0, data, 0, NNS_G2D_VRAM_TYPE_2DSUB, &template, &header);
+    template.position.x = FX32_ONE * 128;
+    template.position.y = FX32_ONE * (0x100 + 96);
+    sceneData->hand1Sprite = CreateSprite(&template);
+    Set2dSpriteAnimActiveFlag(sceneData->hand1Sprite, FALSE);
+    Set2dSpriteVisibleFlag(sceneData->hand1Sprite, FALSE);
+    Set2dSpriteAnimSeqNo(sceneData->hand1Sprite, 0);
+
+    IntroMovie_BuildSpriteResourcesHeaderAndTemplate(0, data, 0, NNS_G2D_VRAM_TYPE_2DMAIN, &template, &header);
+    template.position.x = FX32_ONE * 128;
+    template.position.y = FX32_ONE * 96;
+    sceneData->hand2Sprite = CreateSprite(&template);
+    Set2dSpriteAnimActiveFlag(sceneData->hand2Sprite, FALSE);
+    Set2dSpriteVisibleFlag(sceneData->hand2Sprite, FALSE);
+    Set2dSpriteAnimSeqNo(sceneData->hand2Sprite, 1);
+
+    IntroMovie_BuildSpriteResourcesHeaderAndTemplate(0, data, 0, NNS_G2D_VRAM_TYPE_2DMAIN, &template, &header);
+    template.position.x = FX32_ONE * 128;
+    template.position.y = FX32_ONE * 96;
+    sceneData->sparklesSprite = CreateSprite(&template);
+    Set2dSpriteAnimActiveFlag(sceneData->sparklesSprite, FALSE);
+    Set2dSpriteVisibleFlag(sceneData->sparklesSprite, FALSE);
+    Set2dSpriteAnimSeqNo(sceneData->sparklesSprite, 2);
+
+    for (u8 i = 0; i < 3; ++i) {
+        IntroMovie_BuildSpriteResourcesHeaderAndTemplate(monSpriteResIds[i], data, 0, NNS_G2D_VRAM_TYPE_2DMAIN, &template, &header);
+        template.position.x = FX32_ONE * 128;
+        template.position.y = FX32_ONE * 96;
+        sceneData->starterSprites[i] = CreateSprite(&template);
+        Set2dSpriteAnimActiveFlag(sceneData->starterSprites[i], FALSE);
+        Set2dSpriteVisibleFlag(sceneData->starterSprites[i], FALSE);
+        Set2dSpriteAnimSeqNo(sceneData->starterSprites[i], 0);
+    }
+}
+
+static u32 IntroMovie_Scene4_TexAlloc(u32 size, BOOL is4x4comp) {
+    NNSGfdTexKey key = NNS_GfdAllocTexVram(size, is4x4comp, 0);
+    sub_02015354(key);
+    GF_ASSERT(key != NNS_GFD_ALLOC_ERROR_TEXKEY);
+    return NNS_GfdGetTexKeyAddr(key);
+}
+
+static u32 IntroMovie_Scene4_PlttAlloc(u32 size, BOOL is4pltt) {
```

Comment:
is4pltt?

Reply:
blame nintendo

Reply:
wdym blame nintendo? it's a bit of an odd name, if it is intended to be a 4bpp palette it could use another name tbh

Reply:
> wdym blame nintendo?
that's the parameter name used by nitrosystem. in that regard, i don't make the rules

Reply:
sure I guess, I don't think it's the cleanest but I guess it makes sense